src/coreclr/jit/emitxarch.cpp

```diff
@@ -243,6 +253,19 @@ bool emitter::IsDstSrcSrcAVXInstruction(instruction ins) const
     return (flags & INS_Flags_IsDstSrcSrcAVXInstruction) != 0;
 }
 
+// Returns true if the AVX instruction requires 3 operands and writes result
+// to mask register.
+bool emitter::IsThreeOperandInstructionMask(instruction ins) const
+{
+    if (!UseSimdEncoding())
+    {
+        return false;
+    }
```

Comment:
Don't expect you to do this here, since this is consistent with the other methods.

But I think we can in general simplify this and several of the other SIMD only flags to something like:
```
insFlags flags = CodeGenInterface::instInfo[ins];

if ((flags & INS_Flags_Is3OperandInstructionMask) != 0)
{
    assert(UseSimdEncoding());
    return true;
}

return false;
```

The `UseSimdEncoding()` is itself a flag check now and we should never be setting these SIMD only flags on non-SIMD instructions.

Reply:
I'll log an issue for us to look at that as a cleanup item in some follow up PR.